### PR TITLE
chore: retry for client connection in case of failure

### PR DESCRIPTION
**Pull request by yashLadha**

### CODE REVIEW

**Comment by yashLadha** [comment]
@mykola-mokhnach  The purpose of this PR is to retry on command in the case at startup, RPC client is not connected. Juts a safety check to ensure as we are doing retries at various parts in appium and this practice can be seen in other frameworks as well.

File: lib/remote-debugger.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ const SELECT_APP_RETRY_SLEEP_MS = 500;
 const REMOTE_DEBUGGER_PORT = 27753;
 const SAFARI_BUNDLE_ID = 'com.apple.mobilesafari';
 const BLANK_PAGE_URL = 'about:blank';
+const WDA_CONNECT_RETRIES = 1;
```
Comment (imurchie) [inline lib/remote-debugger.js]: What is `WDA` here?

File: lib/remote-debugger.js [hunk 2]
```diff
@@ -406,9 +407,22 @@ class RemoteDebugger extends events.EventEmitter {
     log.debug(`Selected page after ${timer.getDuration().asMilliSeconds.toFixed(0)}ms`);
   }
 
+  async retryConnector () {
+    for (let retryCount = 0; retryCount < WDA_CONNECT_RETRIES; ++retryCount) {
+      await this.connect(1);
```
Comment (imurchie) [inline lib/remote-debugger.js]: Why a timeout of 1ms?

Comment (yashLadha) [reply lib/remote-debugger.js]: To basically halt for a minimal time, as `0` is also not executed immediately in some cases.

**Comment by mykola-mokhnach** [comment]
> The purpose of this PR is to retry on command in the case at startup, RPC client is not connected. Juts a safety check to ensure as we are doing retries at various parts in appium and this practice can be seen in other frameworks as well.

Does it really help? Could you please provide some log where the retry really helps to recover from an error? In my experience retries could make the code really slow, especially if they are used in frequently called areas. Perhaps, the original issue could (should) be fixed in some other way?

**Comment by yashLadha** [comment]
Closing for now.

**Pull request closed without merging**